src/Launcher/MapMigrations/UnitTooltipBasicMigration.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using War3Api.Object;
+using War3Net.Build;
+
+namespace Launcher.MapMigrations;
+
+/// <summary>
+/// Sets the "Text - Tooltip - Basic", "Text - Tooltip - Revive", and "Text - Tooltip - Awaken" fields for units.
+/// </summary>
+public sealed class UnitTooltipBasicMigration : IMapMigration
+{
+  /// <inheritdoc />
+  public void Migrate(Map map, ObjectDatabase objectDatabase)
+  {
+    var units = objectDatabase.GetUnits();
+    var copiedUnits = units.ToList();
```

Comment:
Is the intention here to materialize a LINQ sequence, or is it meant to create independent copies of the units?
```suggestion
    var units = objectDatabase.GetUnits().ToList();
```

Reply:
Accepted the suggestion.
Yeah, it's to create independent copies of the units. Drake53 explained to me why in his Hive Workshop post about C# but I didn't entirely understand - it just doesn't work without the `ToList`.

Reply:
I tried getting rid of the `ToList` and it worked just fine so... I don't know. Removed it, no more materialization.